src/lib/str.c

```diff
@@ -193,7 +193,7 @@ void xdebug_str_free(xdebug_str *s)
 char *xdebug_sprintf(const char* fmt, ...)
 {
 	char   *new_str;
-	int     size = 1;
+	int     size = 32;
```

Comment:
1 is definitely/measureably impact the performace!

16 - 64 seems to be a good value, will benchmark further

if we do not have to return a value, I think we can define with macro without vargs

Reply:
Graph from various ini sized across 950 measurements:

![image](https://user-images.githubusercontent.com/2228672/87992116-de0ef600-cae7-11ea-9bd2-ff07a6ba2f57.png)

Reply:
total time with custom file buffering vs. without:

![image](https://user-images.githubusercontent.com/2228672/87993062-e36d4000-cae9-11ea-84d3-09def976b7bb.png)

Reply:
I'm okay with increasing that number, but is your profile graph only for running the profiler, or a combination of running all the Xdebug tests? xdebug_str is used quite a lot. I would rather have change this separated out of this PR too.

Reply:
I plan to finish this PR after the PR is merged. With this number I came maybe to a better idea - what about 4k internal buffer to which we will try to print first - if it succeed, we copy the printed length, if not, we loop like before.

So no min 32B memory will be allocated and I think this would be also the fastest, do you like this idea (one global/TS buffer)?

Reply:
> I would rather have change this separated out of this PR too.

PR created - https://github.com/xdebug/xdebug/pull/613